### DISCUSSION TITLE
Design Inquiry: Is there a mobile app design system?

### DISCUSSION THREAD

**Discussion by jsgoudy** [uswds/uswds#4911] [Q&A]
I am working on a design challenge during the interview process and one of the questions concerns choosing a design system for a hypothetical government app. The design guide on USWDS site is for websites and not apps. Is there a design guide specifically for mobile apps? I understand that you can design a PWS or RWS but my assignment specifically concerns designing and building out an app. Any help or guidance would be appreciated. Thanks!

**Comment by sknep** [comment]
There aren't separate/derivative USWDS component libraries or design systems for iOS, Android, and other platforms, nor are they a priority on the [USWDS roadmap](https://github.com/orgs/uswds/projects/13/views/1), but Digital.gov has a list of resources available for mobile experiences here: https://digital.gov/resources/mobile-user-experience-guidelines/ While not a Design System, they may be of some use to you.

**Comment by jcklpe** [comment]
A good design system should be flexible and responsive enough that it can be used in both mobile and desktop, IMO

**Comment by ryparker** [comment]
USWDS itself provides components for the web: HTML, CSS, and JavaScript. It does not ship a native SwiftUI/UIKit or Android Compose component library. Responsive web components can work on a phone, but that is different from a native mobile design system.

For a native app, use the platform's component and interaction guidance as the starting point: [Apple's Human Interface Guidelines](https://developer.apple.com/design/human-interface-guidelines) or [Android's UI design guidance](https://developer.android.com/design/ui). USWDS principles, content guidance, and appropriate brand tokens can inform the experience, while native controls follow their platform's conventions.

A web view or web app can use USWDS directly, but that is a different implementation choice from native platform controls.
